### PR TITLE
Worksheet for beta

### PR DESCRIPTION
**Pull request by itsjoshthedeveloper**

### CODE REVIEW

**Comment by itsjoshthedeveloper** [comment]
don't merge just yet! i'm working on a tutorial for the worksheet too

File: frontend/src/searchContext.tsx [hunk 1]
```diff
@@ -96,18 +105,38 @@ type Store = {
 const SearchContext = createContext<Store | undefined>(undefined);
 SearchContext.displayName = 'SearchContext';
 
+// Calculate upcoming season
+const dt = DateTime.now().setZone('America/New_York');
```
Comment (kevinhu) [inline frontend/src/searchContext.tsx:109]: Should we also handle summer courses here?

File: frontend/src/searchContext.tsx [hunk 2]
```diff
@@ -31,6 +38,8 @@ export type Option = {
   numeric?: boolean;
 };
 
+export const isOption = (x: any): x is Option => 'label' in x && 'value' in x;
```
Comment (kevinhu) [inline frontend/src/searchContext.tsx:41]: Why do we need `x: any` followed by `x is Option`  here? Can we use a type besides `any`?

**Review by kevinhu — APPROVED** [review]
Looks good – this diff is way too large for me to have a sense of any architectural issues but it otherwise checks out.

File: frontend/src/searchContext.tsx (hunk 1 above)
Comment (itsjoshthedeveloper) [reply frontend/src/searchContext.tsx:109]: i thought about that but assumed most people don't do summer session. think we should?

File: frontend/src/searchContext.tsx (hunk 2 above)
Comment (itsjoshthedeveloper) [reply frontend/src/searchContext.tsx:41]: this is a type guard that we can use to check if any value is of type Option

**Pull request merged**